### PR TITLE
Add MyPY to lint GHA workflow

### PR DESCRIPTION
**Pull request by malfet**
Also disable test_run_mypy from  test_type_hints.py as it is running as part of GHA

### CODE REVIEW

**Review by ezyang — APPROVED** [review]
TBH, it's probably safe to just delete the test entirely

**Review by xuzhao9 — APPROVED** [review]

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -213,3 +213,38 @@ jobs:
...
+        with:
+          python-version: 3.8
+          architecture: x64
+      - name: Fetch PyTorch
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Get HEAD commit SHA
+        run: echo ::set-output name=commit-sha::$(git rev-parse HEAD)
+        id: get-commit-sha
```
Comment (samestep) [inline .github/workflows/lint.yml:231]: This step is unnecessary:
```suggestion
```

**Review by samestep — APPROVED** [review]

File: test/test_type_hints.py [hunk 2]
```diff
@@ -128,6 +128,7 @@ def test_doc_examples(self):
                 self.fail(f"mypy failed:\n{stdout}")
 
     @unittest.skipIf(not HAVE_MYPY, "need mypy")
+    @unittest.skipIf(os.getenv("IN_CI") is not None, "Already running in CI during lint")
```
Comment (walterddr) [inline test/test_type_hints.py:131]: reran this locally on base commit shown 1282 file touched instead of 1281. wondering if we missed something.

**Review by janeyx99 — APPROVED** [review]

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -213,3 +213,38 @@ jobs:
...
+        run: |
+          set -eux
+          pip install -r requirements.txt
+          pip install mypy==0.770
+      - name: Run autogen
+        run: |
+          set -eux
+          time python -mtools.generate_torch_version --is_debug=false
+          time python -mtools.codegen.gen -s aten/src/ATen -d build/aten/src/ATen
+          time python -mtools.pyi.gen_pyi --native-functions-path aten/src/ATen/native/native_functions.yaml --deprecated-functions-path "tools/autograd/deprecated.yaml"
```
Comment (ezyang) [inline .github/workflows/lint.yml:242]: For future work; I'd love it if we can consolidate this into a single command to run

File: test/test_type_hints.py (hunk 2 above)
Comment (malfet) [reply test/test_type_hints.py:131]: Thank you for pointing it out. Will do the comparison in the followup PR.

File: .github/workflows/lint.yml (hunk 1 above)
Comment (malfet) [reply .github/workflows/lint.yml:231]: Will do in followup (to avoid waiting once again for internal CI)

**Comment by malfet** [comment]
@ezyang I think it's good to keep the test in place, so that people can validate it locally if they choose to.

**Comment by ezyang** [comment]
> @ezyang I think it's good to keep the test in place, so that people can validate it locally if they choose to.

Sure, local validation is a good idea. I guess I'm just kind of hoping that in the glorious future you can just do something like `python setup.py codegen; mypy` and that's literally all you need to do to validate.

**Pull request closed without merging**